pkg/datapath/loader/base.go

```diff
@@ -400,6 +387,34 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 	args[initArgHostDev1] = hostDev1.Attrs().Name
 	args[initArgHostDev2] = hostDev2.Attrs().Name
 
+	if option.Config.EnableIPv6 {
+		args[initArgIPv6NodeIP] = node.GetIPv6().String()
+		// Docker <17.05 has an issue which causes IPv6 to be disabled in the initns for all
+		// interface (https://github.com/docker/libnetwork/issues/1720)
+		// Enable IPv6 for now
+		sysSettings = append(sysSettings,
+			sysctl.Setting{Name: "net.ipv6.conf.all.disable_ipv6", Val: "0", IgnoreErr: false})
+		localIPv6CIDR := node.GetIPv6AllocRange()
+
+		ciliumHostV6Addrs, err := netlink.AddrList(hostDev1, netlink.FAMILY_V6)
+		if err != nil {
+			return err
+		}
+		for _, addr := range ciliumHostV6Addrs {
+			// Delete cilium_host v6 addresses allocated from IPAM to make sure downgrade from 1.14 to 1.13 works:
+			// https://github.com/cilium/cilium/issues/25938
+			if localIPv6CIDR.Contains(addr.IPNet.IP) {
```

Comment:
One more question - is there any other way to tell that IP was allocated from the IPAM? Asking, as with the multi-pool IPAM, ENI IPAM, etc. we cannot assume that the IPs will be allocated from a single CIDR.

Reply:
Maybe we can call this function to check all the CIDRs: 
https://github.com/cilium/cilium/blob/592777da560bea7838b99223386e943c08d5d052/pkg/node/types/node.go#L578